src/Schema/Directives/WhereAuthDirective.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Illuminate\Contracts\Auth\Factory as AuthFactory;
+use Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective;
+use Nuwave\Lighthouse\Support\Contracts\FieldBuilderDirective;
+
+class WhereAuthDirective extends BaseDirective implements FieldBuilderDirective
+{
+    /**
+     * @var \Illuminate\Contracts\Auth\Factory
+     */
+    protected $authFactory;
+
+    public function __construct(AuthFactory $authFactory)
+    {
+        $this->authFactory = $authFactory;
+    }
+
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Filter a type to only return instances owned by the current user.
+"""
+directive @whereAuth(
+  """
+  Specify the relationship to check for auth
+  """
+  relation: String = "user"
```

Comment:
I am on the fence about the default, maybe it should just be a required argument without default.

Reply:
Sure, I agree. I would prob. also in many cases call that relationship owner or something like that.